docs/source/app_developers_guide/zmq_event_subscription.rst

```diff
@@ -0,0 +1,309 @@
+        content=request)
+
+    # Send the request
+    socket.send_multipart([msg.SerializeToString()])
+
+.. note::
+
+   For information about ``ClientEventsSubscribeRequest``, see
+   :doc:`about_event_subscriptions`.
+
+
+Step 3: Receiving the Response
+==============================
+
+After submitting the subscription request, wait for a response from the
+validator. The validator will return a response indicating whether the
+subscription was successful.
+
+The following example receives the response and verifies the status.
+
+.. code-block:: python
+
+    # Receive the response
+    resp = socket.recv_multipart()[-1]
+
+    # Parse the message wrapper
+    msg = Message()
+    msg.ParseFromString(resp)
+
+    # Validate the response type
+    if msg.message_type != CLIENT_EVENTS_SUBSCRIBE_RESPONSE:
+        print("Unexpected message type")
+
+    # Parse the response
+    response = ClientEventsSubscribeResponse()
+    response.ParseFromString(msg.content)
+
+    # Validate the response status
+    if response.status != ClientEventsSubscribeResponse.OK:
+      print("Subscription failed: {}".format(response.response_message))
+
+
+Step 4: Listening for Events
+============================
+
+After the event subscription request has been sent and accepted, events will
+arrive on the ZMQ socket. The application must start listening for these events.
+
+.. note::
+
+   In order to limit network traffic, individual events are wrapped in an
+   event list message before being sent.
+
+The following example listens for events and prints them indefinitely.
+
+.. code-block:: python
+
+    while True:
+      resp = socket.recv_multipart()[-1]
+
+      # Parse the message wrapper
+      msg = Message()
+      msg.ParseFromString(resp)
+
+      # Validate the response type
+      if msg.message_type != CLIENT_EVENTS:
+          print("Unexpected message type")
+
+      # Parse the response
+      events = EventList()
+      events.ParseFromString(msg.content)
+
+      for event in events:
+        print(event)
+
+
+.. _correlate-events-to-blocks-label:
+
+Correlating Events to Blocks
+============================
+
+An event originates from a specific block. That is, an event is sent to the
+subscriber only when the block is committed and state is updated. As a result,
+events can be treated as output from processing and committing blocks.
+
+An application can subscribe to both ``sawtooth/block-commit`` and
+``sawtooth/state-delta`` events to match state changes with the block
+in which the changes occurred.
+
+All lists of ``block-commit`` events received from the validator will contain a
+``block-commit`` event for the block that the events came from.
+
+.. important::
+
+   For forking networks, we recommend subscribing to ``block-commit`` events
+   in order to watch for network forks and react appropriately. Without
+   a subscription to ``block-commit`` events, there is no way to determine
+   whether a fork has occurred.
```

Comment:
Suggest adding, "It may also be a good idea to only react to events after some number of blocks have been committed on the given fork, when there is some confidence that you are not going to have to revert those changes because you switched to a different fork."

Reply:
Done.